front/src/applications/stdcm/views/StdcmRequestModal.tsx

```diff
@@ -48,71 +54,77 @@ export default function StdcmRequestModal(props) {
   const stdcmRequest = async () => {
-    const params = formatStdcmConf(dispatch, setFailure, t, osrdconf);
-    return postStdcm(params).unwrap();
+    const params = formatStdcmConf(dispatch, setFailure, t, osrdconf as OsrdStdcmConfState);
+    return postStdcm(params as PostStdcmApiArg).unwrap();
   };
 
   useEffect(() => {
     if (currentStdcmRequestStatus === STDCM_REQUEST_STATUS.pending && timetableId) {
       stdcmRequest()
         .then((result) => {
           setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.success);
-          dispatch(updateItinerary(result.path));
-
-          const fakedNewTrain = {
-            ...cloneDeep(result.simulation),
-            id: 1500,
-            isStdcm: true,
-          };
-
-          fakedNewTrain.base.stops = fakedNewTrain.base.head_positions[0].map(
-            (headPosition, index) => ({
-              id: fakedNewTrain.id,
-              name: `stop ${index}`,
-              time: headPosition.time,
-              position: headPosition.position,
-              duration: 0,
-            })
-          );
-
-          getTrainScheduleResults({
-            timetableId,
-            pathId: result.path.id,
-          })
-            .unwrap()
-            .then((timetableTrains) => {
-              const trains = [...timetableTrains, fakedNewTrain];
-              const consolidatedSimulation = createTrain(
-                dispatch,
-                KEY_VALUES_FOR_CONSOLIDATED_SIMULATION,
-                trains,
-                t
-              );
-              dispatch(updateConsolidatedSimulation(consolidatedSimulation));
-              dispatch(updateSimulation({ trains }));
-              dispatch(updateSelectedTrainId(fakedNewTrain.id));
-
-              dispatch(
-                updateSelectedProjection({
-                  id: fakedNewTrain.id,
-                  path: result.path,
-                })
-              );
-
-              dispatch(updateMustRedraw(true));
+          if ('path' in result && 'simulation' in result) {
+            dispatch(updateItinerary(result.path));
+
+            const fakedNewTrain = {
+              ...cloneDeep(result.simulation),
+              id: 1500,
+              isStdcm: true,
+            };
+
+            fakedNewTrain.base.stops = fakedNewTrain.base.head_positions[0].map(
+              (headPosition: { position: number; time: number }, index: number) => ({
+                id: fakedNewTrain.id.toString(),
+                name: `stop ${index}`,
+                time: headPosition.time,
+                position: headPosition.position,
+                duration: headPosition.time,
+                line_code: 0,
+                line_name: '',
+                track_number: 0,
+                track_name: '',
+              })
+            );
+
+            getTrainScheduleResults({
+              timetableId,
+              pathId: result.path?.id as number,
```

Comment:
No need for the cast and the ?
```suggestion
              pathId: result.path.id,
```

Reply:
done